lib/private/DB/ConnectionAdapter.php

```diff
@@ -172,12 +171,21 @@ public function close(): void {
 		$this->inner->close();
 	}
 
+	/**
+	 * @param mixed $input
+	 * @param int $type
+	 * @deprecated 30.0.0 Only strings are supported as database type in the end and the $type parameter is ignored going forward
+	 */
 	public function quote($input, $type = IQueryBuilder::PARAM_STR) {
-		return $this->inner->quote($input, $type);
+		if ($type !== IQueryBuilder::PARAM_STR) {
+			\OC::$server->getLogger()->debug('Parameter $type is no longer supported and the function only handles resulting database type string', ['exception' => new \InvalidArgumentException('$type parameter is no longer supported')]);
```

Comment:
Please avoid using `getLogger` it’s deprecated.
```suggestion
			\OCP\Server::get(LoggerInterface::class)->debug('Parameter $type is no longer supported and the function only handles resulting database type string', ['exception' => new \InvalidArgumentException('$type parameter is no longer supported')]);
```

Reply:
Also you use $type in single quotes so it will not be interpolated.

Reply:
> Also you use $type in single quotes so it will not be interpolated.

The stacktrace should tell you what the value was. It was never meant to "interpolate"